lib/Common/Core/ConfigParser.cpp

```diff
@@ -337,19 +337,70 @@ void ConfigParser::ParseConfig(HANDLE hmod, CmdLineArgsParser &parser)
         {
             return;
         }
+
         if (_wfopen_s(&configFile, filename, _u("r, ccs=UNICODE")) != 0 || configFile == nullptr)
         {
             return;
         }
     }
+#else
+    // Two-pathed for a couple reasons
+    // 1. PAL doesn't like the ccs option passed in.
+    // 2. _wfullpath is not implemented in the PAL.
+    // Instead, on xplat, we'll check the HOME directory to see if there is
+    // a config file there that we can use
+    if (_wfopen_s(&configFile, filename, _u("r")) != 0 || configFile == nullptr)
+    {
+        WCHAR homeDir[MAX_PATH];
+
+        if (GetEnvironmentVariable(_u("HOME"), homeDir, MAX_PATH) == 0)
+        {
+            return;
+        }
+        
+        WCHAR configFileFullName[MAX_PATH];
+
+        StringCchPrintf(configFileFullName, MAX_PATH, _u("%s/%s.config"), homeDir, _configFileName);
+        if (_wfopen_s(&configFile, configFileFullName, _u("r")) != 0 || configFile == nullptr)
+        {
+            return;
+        }
+    }
+#endif
 
-    while (fwscanf_s(configFile, _u("%s"), configBuffer, MaxTokenSize) != FINISHED)
+    char16 configBuffer[MaxTokenSize];
+    int index = 0;
+
+    // We don't expect the token to overflow- if it does
+    // the simplest thing to do would be to ignore the
+    // read tokens
+    // We could use _fwscanf_s here but the function
+    // isn't implemented in the PAL and we'd have to deal with
+    // wchar => char16 impedance mismatch.
+    while (index < MaxTokenSize)
     {
-        if ((err = parser.Parse(configBuffer)) != 0)
+        int curChar = fgetc(configFile);
+        if (curChar == '\n' || curChar == FINISHED)
         {
-            break;
+            configBuffer[index] = 0;
+            if ((err = parser.Parse(configBuffer)) != 0)
+            {
+                break;
+            }
+
+            if (curChar == FINISHED)
+            {
+                break;
+            }
+
+            index = 0;
+        }
+        else
+        {
+            configBuffer[index++] = (char16) curChar;
```

Comment:
I see. so probably we don't even need to put a comment there? i.e. user path is unicode? but I guess there is no path variable within config file?